src/clj/puppetlabs/enterprise/services/puppet_profiler/puppet_profiler_core.clj

```diff
@@ -0,0 +1,132 @@
+(ns puppetlabs.enterprise.services.puppet-profiler.puppet-profiler-core
+  (:import (com.codahale.metrics MetricRegistry Timer)
+           (com.puppetlabs.enterprise MetricsPuppetProfiler)
+           (com.puppetlabs.puppetserver PuppetProfiler))
+  (:require [clojure.string :as str]
+            [schema.core :as schema]
+            [puppetlabs.kitchensink.core :as ks]
+            [clojure.tools.logging :as log]
+            [puppetlabs.trapperkeeper.services.status.status-core :as status-core]
+            [puppetlabs.metrics :as metrics]
+            [puppetlabs.i18n.core :refer [trs]]))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Schemas
+
+(def MetricsProfilerServiceContext
+  {:profiler (schema/maybe PuppetProfiler)})
+
+(def MetricsProfilerServiceConfig
+  {(schema/optional-key :enabled) schema/Bool})
+
+(def PEPuppetProfilerStatusV1
+  {(schema/optional-key :experimental)
+   {:function-metrics [{:function schema/Str
+                        :count schema/Int
+                        :mean schema/Num
+                        :aggregate schema/Num}]
+    :resource-metrics [{:resource schema/Str
+                        :count schema/Int
+                        :mean schema/Num
+                        :aggregate schema/Num}]
+    :catalog-metrics [{:metric schema/Str
+                       :count schema/Int
+                       :mean schema/Num
+                       :aggregate schema/Num}]
+    :inline-metrics [{:metric schema/Str
+                      :count schema/Int}]}})
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Private
+
+(schema/defn metrics-profiler :- PuppetProfiler
```

Comment:
@adrienthebo Good question.  I swear we had documented better at some point why we do this but am not finding any references to this now.  I believe the reasons we originally started doing this in the Puppet Server-related projects instead of using `defn-` or a `^:private` modifier are:

1) Makes it easier to reference the function directly from test code when you need to.  You can work around the access modifier in tests by using the [@#'some.ns/var](https://github.com/bbatsov/clojure-style-guide#access-private-var) modifier but that is a bit uglier.

2) We were enamored with the idea of using "pragma" blocks like the ";;; ... Private " above for readability, following the lead of [Prismatic](https://github.com/plumatic/eng-practices/blob/4cfaa9099841c7e041a1eb4f41e66fa460774386/clojure/20130927-ns-organization.md#separate-code-into-pragma-blocks).

In the above case, I was just copying in the code basically as-is from PE, pragma blocks and all.  You'll see this style for the most part in Puppet Server code although it certainly isn't uncommon to see the `defn-` style in other projects.  Mostly a question of style, I guess.